### PR TITLE
Initial release of crossbeam-skiplist

### LINKED ISSUE #635: crossbeam-skiplist 0.1 release
This is a tracking issue for the initial release of crossbeam-skiplist.

## TODO

- [x] Add docs (https://github.com/crossbeam-rs/crossbeam/issues/503#issuecomment-631679442)
  - [x] Crate-level docs; done in #535
  - [x] Docs for `SkipMap`; done in #535
  - [x] Docs for `SkipSet`; done in #655
  - [x] Docs for `SkipList`; done in #655
- [x] Update readme; done in #887

## Open issues

- [ ]  Remove the `'static` bounds (#205)
- [ ] `Entry`'s name is misleading (https://github.com/crossbeam-rs/crossbeam/pull/535#issue-440783380); maybe good to rename to [`Cursor`](https://doc.rust-lang.org/stable/std/collections/linked_list/struct.Cursor.html)? (https://github.com/crossbeam-rs/crossbeam/pull/535#issuecomment-651188964)
- [x] `set::Entry` could implement `Deref<Target=T>` for convenience. (https://github.com/crossbeam-rs/crossbeam/pull/535#issue-440783380); done in #657
- memory leaks 
  - [ ] SkipList #614
  - [x] Skip{Map,Set} #671; fixed in #673
  - [x] Skip{Map,Set} #672; fixed in #673 
- [ ] re-export in crossbeam crate (#889)

### PR DESCRIPTION
**Pull request by taiki-e**
Now that all the TODOs in #635 have been resolved, I think we can release the initial version.
There are still some open issues in #635, but I don't think they are blockers of the 0.1 release.

(There are some breaking changes in the master branch, but skiplist is not affected.)

Closes #635

### CODE REVIEW

**Comment by taiki-e** [comment]
yeah, this is ready to review, as PR is not marked as a draft.

**Comment by emilHof** [comment]
is there anything holding it back from having the `0.0.1` release now, or is it just waiting on a core contributor to approve it?

**Review by Amanieu — APPROVED** [review]

**Comment by taiki-e** [comment]
bors r=Amanieu

**Comment by taiki-e** [comment]
Published https://crates.io/crates/crossbeam-skiplist/0.1.0

**Comment by emilHof** [comment]
yay its done awesome!
now that `0.0.1` is out, what are some things that might be good to look at for upgrades/fixes in the `skiplist` crate?

**Pull request merged**
